parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -266,10 +270,11 @@ async def get_path_at_timestamp(self, entry_id: EntryID, timestamp: Pendulum) ->
         # Loop over parts
         parts = []
         while not is_workspace_manifest(current_manifest):
-
             # Get the manifest
             try:
+                current_manifest = cast(Union[FolderManifest, FileManifest], current_manifest)
```

Comment:
why the cast ?
couldn't we just do `current_manifest: FolderishManifest, _ = await self.load(...` ?

Reply:
no, there we need explicitly to point for the entire loop that `current_manifest` is not a `WorkspaceManifest` (otherwise `current_manifest.parent` is invalid.)